saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -70,6 +81,25 @@ class Meta:
         error_type_class = ProductError
         error_type_field = "product_errors"
 
+    @classmethod
+    def perform_mutation(cls, _root, info, ids, **data):
+        response = super().perform_mutation(
+            _root,
+            info,
+            ids,
+            manager=info.context.plugins,
+            **data,
+        )
+        return response
+
+    @classmethod
+    def bulk_action(cls, queryset, manager):
+        collections_ids = queryset.values_list("id", flat=True)
+        products = list(models.Product.objects.filter(collections__in=collections_ids))
```

Comment:
IMO, you need distinct here.
![image](https://user-images.githubusercontent.com/23282005/112161580-94812880-8beb-11eb-89a5-e64bc5211576.png)